README.md

```diff
@@ -130,11 +130,43 @@ There are two main types of documentation in this repository:
 
 ## Performance
 
-Benchmarks taken on `Intel(R) Core(TM) i5-7300HQ CPU @ 2.50GHz`
-For a proof-size of `2^16` constraints/gates:
+Benchmarks taken on `Intel(R) Xeon(R) CPU E5-1620 v4 @ 3.50GHz`.
 
-- Proving time: `5.46s`
-- Verification time: `6.13ms`. **(This time will not vary depending on the circuit-size.)**
+All results are presented on a min/avg/max time based on the number of constraints performed in a circuit. None of the tests used public inputs and its number may affect the verification time.
+
+#### Proving performance
+```
+2^5  = 32 constraints      time:   [31.788 ms 31.825 ms 31.862 ms]
+2^6  = 64 constraints      time:   [41.388 ms 41.429 ms 41.471 ms]
+2^7  = 128 constraints     time:   [63.496 ms 63.539 ms 63.573 ms]
+2^8  = 256 constraints     time:   [100.32 ms 100.45 ms 100.68 ms]
+2^9  = 512 constraints     time:   [140.36 ms 140.58 ms 140.78 ms]
+2^10 = 1024 constraints    time:   [256.18 ms 256.48 ms 256.83 ms]
+2^11 = 2048 constraints    time:   [478.55 ms 480.41 ms 482.70 ms]
+2^12 = 4096 constraints    time:   [772.29 ms 776.97 ms 781.71 ms]
+2^13 = 8192 constraints    time:   [1.3406 s 1.3448 s 1.3494 s]
+2^14 = 16384 constraints   time:   [2.6250 s 2.6316 s 2.6379 s]
+2^15 = 32768 constraints   time:   [5.1982 s 5.2228 s 5.2460 s]
+2^16 = 65536 constraints   time:   [9.2356 s 9.3039 s 9.3636 s]
+2^17 = 131072 constraints  time:   [17.836 s 17.883 s 17.936 s]
+```
+
+#### Verification performance
+```
+2^5  = 32 constraints      time:   [15.287 ms 15.700 ms 16.066 ms]
+2^6  = 64 constraints      time:   [15.172 ms 15.438 ms 15.657 ms]
+2^7  = 128 constraints     time:   [15.335 ms 15.666 ms 15.996 ms]
+2^8  = 256 constraints     time:   [14.988 ms 15.116 ms 15.383 ms]
+2^9  = 512 constraints     time:   [15.047 ms 15.181 ms 15.471 ms]
+2^10 = 1024 constraints    time:   [15.630 ms 15.888 ms 16.303 ms]
+2^11 = 2048 constraints    time:   [15.524 ms 15.719 ms 15.913 ms]
+2^12 = 4096 constraints    time:   [15.178 ms 15.445 ms 15.593 ms]
+2^13 = 8192 constraints    time:   [15.437 ms 15.874 ms 16.553 ms]
+2^14 = 16384 constraints   time:   [16.059 ms 16.386 ms 16.608 ms]
+2^15 = 32768 constraints   time:   [15.969 ms 16.357 ms 16.660 ms]
+2^16 = 65536 constraints   time:   [16.233 ms 16.462 ms 16.639 ms]
+2^17 = 131072 constraints  time:   [18.213 ms 18.680 ms 18.886 ms]
+```
```

Comment:
Could you run again the benches after the lates additions and add back the new results?